python/cuml/dask/ensemble/base.py

```diff
@@ -0,0 +1,179 @@
+import dask
+import math
+
+from cuml.dask.common.input_utils import DistributedDataHandler, \
+    wait_and_raise_from_futures, concatenate
+from dask.distributed import default_client
+
+
+class BaseRandomForestModel(object):
+    """
+    BaseRandomForestModel defines functions used in both Random Forest
+    Classifier and Regressor for Multi Node and Multi GPU models. The common
+    functions are defined here and called from the main Random Forest Multi
+    Node Multi GPU APIs. The functions defined here are not meant to be used
+    as a part of the public API.
+    """
+
+    def _create_model(self, model_func,
+                      client,
+                      workers,
+                      n_estimators,
+                      base_seed,
+                      **kwargs):
+
+        self.client = default_client() if client is None else client

```

Comment:
Very minor- but since you are consolidating, I think you should make use of `get_client(client)` in `cuml.dask.common.utils`